### PR TITLE
Pubby class remake

### PR DESCRIPTION
**Pull request by spockye**
## About The Pull Request

Replaces the Current Pubby class carrier with a new version. 
image

## Why It's Good For The Game

the current Pubby class carrier is outdated, and barely works. this version brings it up to speed, and makes it more tolerable

****
## Changelog
:cl:
add: Remade the pubby class light carrier
/:cl:

### CODE REVIEW

**Comment by MusselMain** [comment]
Nice! I'm all for making shiptests feel more lived in and realistic (at least in a 'this is a long range spaceship that probably needs beds or bunkbeds for their crew')

**Comment by Apogee-dev** [comment]
Looks like you still need fire alarms and firelocks. Also, you should probably change the miner outfit to the Asteroid Miner one, since the default for miner is Nanotrasen-themed. Same goes for any miner hardsuits you might have on the map, use the indie one instead. I really need to update the miner outfits to just make indie the default… You also might want to add two Deckhand slots or so. The Pubby’s a small ship, but a default limit of four players is small even for that.

Other than that the only other changes I’d suggest are adding a prefix and namelists in the config.

**Comment by spockye** [comment]
alright, it should be fine and Dandy now. 
---
the crew will now be:
1x captain
1x asteroid miner
1x engineer
2x pilots
____
fire alarms and firelocks have been added, the hardsuits and shaft miner are now independent, a independent prefix has been added, and a namelist

File: _maps/configs/pubby.json [hunk 1]
```diff
@@ -1,15 +1,22 @@
 {
 	"map_name": "Pubby-class Light Carrier",
+	"prefix": "ISV",
+	"namelists": ["GENERAL", "BRITISH_NAVY", "MYTHOLOGICAL", "WEAPONS"],
 	"map_short_name": "Punny-class",
```
Comment (Apogee-dev) [inline _maps/configs/pubby.json]: "map_short_name": "Pubby-class",

while amusing, the shortname actually shows up in the ship purchase menu and a few other places and should actually match the ship's name.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

File: _maps/configs/pubby.json (hunk 1 above)
Comment (spockye) [reply _maps/configs/pubby.json]: oh,pft i didnt set it to be like that. it's always been that way

**Comment by Latentish** [comment]
I like this pubby more than the old one.

**Comment by PrefabQuasar** [comment]
agreed, this looks like a really good remap

**Comment by Derpguy3** [comment]
Space pods need an air canister to use for cabin internals, make sure there is one for the other frame

**Comment by Apogee-dev** [comment]
Can you add shutters for the windows and engines? Make sure they start out closed so players can't miss the fact that they're there.

**Comment by spockye** [comment]
oh god why is it all failing

**Comment by Apogee-dev** [comment]
> oh god why is it all failing

the .dmm isn't parsing, something fucked up in the file itself. It could be that you accidentally left a string open or added duplicate quotes somewhere when adding the shutters?

File: _maps/configs/pubby.json [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+			"officer": true,
+			"slots": 1
+		},
+		"Shaft Miner": 1,
+		"Engineer": {
+			"outfit": "/datum/outfit/job/engineer/nt",
+			"slots": 1
+		},
+		"Pilot": {
+			"outfit":"/datum/outfit/job/cargo_tech/pilot",
```
Comment (Apogee-dev) [inline _maps/configs/pubby.json]: You'll want to make a new outfit datum here, this one appears to have been made for the Tartarus or some other ship and isn't really NT themed. My recommendation would be to make it security/nt/pilot, or to use the normal security/nt outfit while keeping the Pilot job title.

Some of the locker contents are also mismatched with the new Nanotrasen theme. Use under/rank/security/officer/nt for the officer jumpsuits in the pilot locker (bluesec shirts are the indie ones) and ditch the dusters, they're a strongly indie flavor. For the firearm locker, replace the M1911s and their magazines with Commanders and maybe give the captain an egun if you want his gun to be different from the rest of the crew.

You also still have independent mining hardsuits in the SSUs, you can replace those with normal mining suits.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

**Review by Apogee-dev — APPROVED** [review]

**Comment by PrefabQuasar** [comment]
couple of notes on this:
it would be nice if you expanded the rear airlock to let people drag crates and etc in with them
the picture shown has buttons and fire alarms not properly rotated to fit the walls, if you've already fixed that, ignore this
consider adding light switches

**Comment by Apogee-dev** [comment]
You've got a public airlock leading into dorms. Always use the public/glass subtype for these, the basic type is, well. ugly as sin.

**Comment by Derpguy3** [comment]
A few things I had noticed while testing out the remake.

- Pod hanger is missing an air alarm.
- Missing a pod circuit and pod core to finish building the second pod.

**Comment by Latentish** [comment]
> A few things I had noticed while testing out the remake.
> 
> * Pod hanger is missing an air alarm.
> * Missing a pod circuit and pod core to finish building the second pod.

We shouldn't have issues marging this once you address this.

**Review by Latentish — APPROVED** [review]
Good changes. I especially like the json changes to bring it up to more recent standards of how these files are written. Looking real margable.

**Pull request merged**